x/distribution/keeper/allocation.go

```diff
@@ -90,8 +102,15 @@ func (k Keeper) AllocateTokens(
 
 		// TODO consider microslashing for missing votes.
 		// ref https://github.com/cosmos/cosmos-sdk/issues/2525#issuecomment-430838701
-		powerFraction := sdk.NewDec(vote.Validator.Power).QuoTruncate(sdk.NewDec(totalPreviousPower))
+		powerFraction := sdk.NewDec(vote.Validator.Power).QuoTruncate(sdk.NewDec(totalPreviousPower - blacklisted_val_power))
 		reward := feesCollected.MulDecTruncate(voteMultiplier).MulDecTruncate(powerFraction)
+
+		valAddr := validator.GetOperator().String()
+		if valAddr == BLACKLISTED_VAL_ADDR {
```

Comment:
```suggestion
		if valAddr == BLACKLISTED_VAL_ADDR {
		    continue
```
Can we add a line like this to the top of the loop instead?